File: easy/balance_string/balance_string.py
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """
        number_of_L = 0
        number_of_R = 0
        balanced_splits = 0

        for c in s:
            if c == 'L':
                number_of_L += 1
            elif c == 'R':
                number_of_R += 1
            if number_of_L == number_of_R:
                balanced_splits += 1

        return balanced_splits


s = Solution()
print("Solution 1 :", s.balancedStringSplit("RLRRLLRLRL"))
print("Solution 2 :", s.balancedStringSplit("RLLLLRRRLR"))
print("Solution 3 :", s.balancedStringSplit("LLLLRRRR"))
print("Solution 4 :", s.balancedStringSplit("RLRRRLLRLL"))

File: easy/array/balance_string/balance_string.go
package main

import "fmt"

func balancedStringSplit(s string) int {
	numberOfL := 0
	numberOfR := 0

	balancedSplits := 0
	for _, c := range s {
		if c == 'L' {
			numberOfL++
		} else if c == 'R' {
			numberOfR++
		}
		if numberOfL == numberOfR {
			balancedSplits++
		}
	}
    return balancedSplits
}

func main() {
	fmt.Println("Solution 1 : ", balancedStringSplit("RLRRLLRLRL"))
	fmt.Println("Solution 2 : ", balancedStringSplit("RLLLLRRRLR"))
	fmt.Println("Solution 3 : ", balancedStringSplit("LLLLRRRR"))
	fmt.Println("Solution 4 : ", balancedStringSplit("RLRRRLLRLL"))
}
